stix_shifter_modules/msatp/readme.md

```diff
@@ -0,0 +1,423 @@
+# MSATP - microsoft defender for endpoint module
+This module uses KQL to query the EDR API.
+
+Queries are constructed as unions for all the relevant event tables based on the queried fields.
+The potential tables are `DeviceProcessEvents`, `DeviceNetworkEvents`, 
+`DeviceRegistryEvents`, `DeviceFileEvents`, `DeviceImageLoadEvents` and `DeviceEvents`
+
+There are 3 new configuration properties (default to `false`):
+### `includeNetworkInfo`:
+If true, every query is joined with `DeviceNetworkInfo` table in order to get the active network adapters from the relevant time of\
+the event (IP and mac addresses).
+
+### `includeHostOs`:
+If true, every query is also joined with `DeviceInfo` table to get the host operating system.
```

Comment:
i hit into an error while running a query by setting `includeHostOs = true`-

`{'success': False, 'connector': 'msatp', 'error': "msatp connector error => {'error': {'code': 'BadRequest', 'message': 'Query execution has exceeded the allowed limits. The query execution was preempted. This could possibly be due to high CPU and/or memory resource consumption. Optimize your query by following best practices and try again.', 'target': '<code>'}}", 'code': 'unknown'}`

Default timeout runs into timeout error. I got the above error after increasing the timeout=60. 

So I'm wondering is there any certain limitation on using these parameters? Is there any setting need to be enabled on the api side to use these query options?

Reply:
So as stated in the documentation turning on these options will have higher cpu usage in msatp api - which has cpu limits. so in case of a broad query you might exceed the cpu qouta. This is why these options are optional and can be turned on/off per use case.
having said that - in our tests with a real customer environment in real investigations none of the queries resulted in exceeding the limit (and have been running for months like this).
probably your query was very general, yeliding a big number of rows.
that is more common for threat hunting use cases, and for those cases one can use the connector with all options turned off.

Reply:
thanks for the details.